File: civitasruby/pkg/Civitas-0.0.1/lib/sorpresa_pagar_cobrar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class SorpresaPagarCobrar < Sorpresa
  def initialize(texto,valor)
    super(texto)
    @valor = valor
  end
  
  def aplicar_a_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        super(actual,todos)
        todos[actual].modificar_saldo(@valor)
      end
  end
  
  def to_string
    a = "Pagar-Cobrar"
    return a
  end
  public_class_method :new
end
end

File: civitasjava/src/Civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;


public class SorpresaPagarCobrar extends Sorpresa{
    private int valor;
    
    SorpresaPagarCobrar(int valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
    
    @Override
    public String toString(){
        return "SorpresaPagarCobrar{ Valor: " +valor +super.toString() +"}";
    }
}
